benchmarks/src/environment/snake.py

```diff
@@ -0,0 +1,68 @@
+from typing import List, Dict, Any
+
+import dataclasses
+import logging
+import subprocess
+import os
+
+from pathlib import Path
+from . import Environment, EnvironmentDescriptor
+
+
+@dataclasses.dataclass(frozen=True)
+class SnakeClusterInfo:
+    workdir: Path
+
+
+class SnakeEnvironmentDescriptor(EnvironmentDescriptor):
+    def create_environment(self, workdir: Path) -> Environment:
+        info = SnakeClusterInfo(workdir)
+        return SnakeEnvironment(info)
+
+    def name(self) -> str:
+        return "snake"
+
+    def parameters(self) -> Dict[str, Any]:
+        return {}
+
+    def metadata(self) -> Dict[str, Any]:
+        return {}
+
+
+class SnakeEnvironment(Environment):
+    def __init__(self, info: SnakeClusterInfo):
+        self.info = info
+        self.snakefile = "Snakefile"
+        self.state = "initial"
+
+    def __enter__(self):
+        self.start()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.stop()
+
+    @property
+    def workdir(self) -> Path:
+        return self.info.workdir
+
+    def start(self):
+        logging.info("Creating Snakefile")
+        file = open(self.snakefile, "w")
+        file.close()
```

Comment:
```suggestion
        with open(self.snakefile, "w") as f:
            pass
```
But if it's empty, it's useless to create it here.